R/data_read.R

```diff
@@ -300,6 +304,83 @@ data_read <- function(path,
   }
   out <- do.call(rio::import, c(rio_args, list(...)))
 
+  # check if loaded file is a data frame, or not (e.g. model objects)
+  file_is_data <- .loaded_file_is_data(out)
+  if (!isFALSE(file_is_data)) {
+    # if file could be coerced to a data frame, overwrite out
+    out <- file_is_data
+  }
+
+  out
+}
+
+
+.read_base_rda <- function(path, verbose = TRUE, ...) {
+  if (verbose) {
+    insight::format_alert("Reading data...")
+  }
+
+  # check URLs
+  path <- .check_path_url(path)
+  env <- new.env()
+  load(file = path, envir = env)
+
+  # if the RData file contains more than one object, we don't check the output
+  # but just return everything
+  if (length(ls(env)) > 1) {
+    if (verbose) {
+      insight::format_alert("File contained more than one object, returning all objects.")
+    }
+    return(as.list(env))
+  }
+
+  # retrieve loaded object
+  out <- get(ls(env)[1], env)
+
+  # check if loaded file is a data frame, or not (e.g. model objects)
+  file_is_data <- .loaded_file_is_data(out, verbose)
+  if (!isFALSE(file_is_data)) {
+    # if file could be coerced to a data frame, overwrite out
+    out <- file_is_data
+  }
+
+  out
+}
+
+
+.read_base_rds <- function(path, verbose = TRUE, ...) {
+  if (verbose) {
+    insight::format_alert("Reading data...")
+  }
+
+  # check URLs
+  path <- .check_path_url(path)
+  out <- readRDS(file = path)
+
+  # check if loaded file is a data frame, or not (e.g. model objects)
+  file_is_data <- .loaded_file_is_data(out, verbose)
+  if (!isFALSE(file_is_data)) {
+    # if file could be coerced to a data frame, overwrite out
+    out <- file_is_data
+  }
+
+  out
+}
+
+
+# check input helper --------------------------------------------------------
+
+# return an URL object if path is a URL
+.check_path_url <- function(path) {
+  url_pattern <- "^(https?|ftp)://(.*)"
```

Comment:
Does this allow `http`? (not sure if we should)

Reply:
Yes, it does. Not sure if we should not allow `http://`?